models/reviews_models.js

```diff
@@ -90,10 +95,36 @@ exports.updateReviewById = (body, params) => {
       status: 400,
       msg: "Invalid request: Enter a valid order query",
     });
-  } else {
-    queryString += ` ORDER BY ${sort_by} ${order.toUpperCase()}`;
-  }
-  return db.query(queryString).then(({ rows: reviews }) => {
-    return reviews;
-  });
-  };
\ No newline at end of file
+  } 
+  if(category){
+        queryString += `WHERE category = $1`
+        queryValues.push(category);
+      } 
+      queryString += `GROUP BY reviews.review_id`; 
+      queryString += ` ORDER BY ${sort_by} ${order.toUpperCase()}`;
+      return db.query(queryString, queryValues).then(({ rows: reviews }) => {
+        if (!reviews[0]) {
+          return fetchAllCategories().then((category) => {
```

Comment:
it should be 'catergories' that is returned from this map as it's an array of the existing categories, of which there is more than one. Syntactically it feels strange to map over a single thing - which is what your parameter implies